centreon/common/cisco/standard/snmp/mode/stack.pm

```diff
@@ -229,17 +230,22 @@ my $oid_cswRingRedundant = '.1.3.6.1.4.1.9.9.500.1.1.3.0';
 sub manage_selection {
     my ($self, %options) = @_;
     $self->{snmp} = $options{snmp};
+    $self->{ignore_cswringredundant} = $self->{option_results}->{ignore_cswringredundant};
 
-    $self->{global} = { waiting => 0, progressing => 0, added => 0, ready => 0, sdmMismatch => 0, 
-        verMismatch => 0, featureMismatch => 0, newMasterInit => 0, provisioned => 0, 
+    $self->{global} = { waiting => 0, progressing => 0, added => 0, ready => 0, sdmMismatch => 0,
+        verMismatch => 0, featureMismatch => 0, newMasterInit => 0, provisioned => 0,
         invalid => 0, removed => 0 };
     $self->{members} = {};
 
     my $redundant = $self->{snmp}->get_leef(oids => [ $oid_cswRingRedundant ], nothing_quit => 1);
 
     if ($redundant->{$oid_cswRingRedundant} != 1) {
```

Comment:
Now you have a binary option mechanism, you can just transform this loop in something more readable 

```suggestion
    if (($redundant->{$oid_cswRingRedundant} != 1) && (defined($self->{$self->{option_results}->{ignore_cswringredundant)) {
```

You "save" an if loop :)